File: Echo Sender.py
import socket

s = socket.socket()
s.bind(("localhost", 7000))
s.listen(1)

print("Server started. Waiting for client...")
c, a = s.accept()
print("Client connected:", a)

while True:
    msg = c.recv(1024).decode()
    if msg == "exit":
        break
    print("Client:", msg)
    c.send(msg.encode())   # echo back

c.close()
s.close()

File: echoSender.java
import java.net.*;
import java.io.*;

public class echoSender {
    public static void main(String[] args) throws Exception {
        ServerSocket serverSocket = new ServerSocket(1234);
        System.out.println("Echo Server Started...");

        Socket socket = serverSocket.accept();
        BufferedReader in = new BufferedReader(
                new InputStreamReader(socket.getInputStream()));
        PrintWriter out = new PrintWriter(socket.getOutputStream(), true);

        String msg;
        while ((msg = in.readLine()) != null) {
            out.println(msg); // echo back
        }

        socket.close();
        serverSocket.close();
    }
}
